contentcuration/contentcuration/models.py

```diff
@@ -1996,6 +1998,8 @@ class File(models.Model):
     source_url = models.CharField(max_length=400, blank=True, null=True)
     uploaded_by = models.ForeignKey(User, related_name='files', blank=True, null=True, on_delete=models.SET_NULL)
 
+    modified = models.DateTimeField(auto_now=True, verbose_name="modified", null=True)
```

Comment:
This should generally work fine - as new files are created one at a time, and so will have the `modified` set by `auto_now`. File updating is done in bulk, so as far as I can tell this will not update `auto_now` (I think this may be affecting our updates of ContentNodes too - so I might need to file a separate issue for that).

The only thing this might miss is file deletion or node deletion, where node deletion will normally be a move to the trash tree. Wonder if we might have to look at node modified as well?

Reply:
Ah good catch, I definitely overlooked the deletion case. I'm thinking that in the case of node/file deletion/move, we could set the cache modified date to the modified date of the earliest date of the associated files prior to the change, if that time is less than the existing cache modified. So if that deleted item was the most recent, and it was included in calculation previously, that date should still be prior to the cache modified. Otherwise, it would almost always be the case, so triggering an update of the cache. Do you think that can work? Am I overlooking anything in that strategy?

Re file updating: yeah that could definitely be an issue with the `modified` field specifically. It seemed to me that there weren't many cases where a file would be updated in a way that would cause its resource size to change. But rethinking it now, if the file is removed from the content node, surely we would need to trigger change the cache modified time.

Reply:
Also, the existing calculation didn't include assessment items.... that should be included too, correct?

Reply:
After a quick slack convo - I think I know what this means now!

So yes, let's update the cache modified when files are deleted, based on the deleted file's modified date, and take the min of the last modified date for all files when we move a content node.

I think we can skip assessment items for now, to avoid making this PR more unmanageable, but we should be considering them, ideally!